### PR TITLE
Change api to accept header and url versioning.

### PR DESCRIPTION
**Pull request by ghost**
API now allows to specify version either url and header way.
Also:
- v2 version of API identifies related objects by ids (due to `ModelSerializer`). This is the new behaviour.
- v1 version of API identifies related objects by urls (due to `HyperlinkedModelSerializer`). This is the old behaviour.

### CODE REVIEW

File: dnsaas/settings.py [hunk 1]
```diff
@@ -119,6 +119,7 @@ def __getitem__(self, item):
     'django.contrib.auth.middleware.AuthenticationMiddleware',
     'django.contrib.messages.middleware.MessageMiddleware',
     'threadlocals.middleware.ThreadLocalMiddleware',
+    'dnsaas.api.middlewares.VersionSwitch',
```
Comment (szok) [inline dnsaas/settings.py]: dnsaas.api.middleware.VersionSwitch
less "s" ?

File: dnsaas/api/v2/views.py [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+        return Response(data, status=code, headers=headers)
+
+    def update(self, request, *args, **kwargs):
+        partial = kwargs.pop('partial', False)
+        instance = self.get_object()
+        serializer = self.get_serializer(
+            instance, data=request.data, partial=partial,
+        )
+        serializer.is_valid(raise_exception=True)
+        print(serializer.validated_data)
```
Comment (szok) [inline dnsaas/api/v2/views.py]: print :)

File: dnsaas/api/middlewares.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+def get_api_version(request):
+    """Gets api version from header"""
+    value = request.META.get('HTTP_ACCEPT', '')
+    version = re.findall(r'v\d+', value, re.IGNORECASE)
+    version = version[0] if version else None
+    return version
+
+
+class VersionSwitch(object):
```
Comment (mkurek) [inline dnsaas/api/middlewares.py]: could you describe (briefly) how does this work and what is the purpose of this middleware?

File: dnsaas/settings.py [hunk 4]
```diff
@@ -178,7 +179,7 @@ def __getitem__(self, item):
     'DEFAULT_PAGINATION_CLASS':
         'rest_framework.pagination.LimitOffsetPagination',
     'DEFAULT_VERSIONING_CLASS':
-        'rest_framework.versioning.AcceptHeaderVersioning',
+        'rest_framework.versioning.NamespaceVersioning',
```
Comment (mkurek) [inline dnsaas/settings.py:182]: imo we couldn't (for now) drop previous behaviour (header version) and we should create our own versioning system which would combine `AcceptHeaderVersioning` and `NamespaceVersioning` - you could specify either version in url or in header. Calling `/api/<resource>` should recognise version in header, but calling `/api/v<version>/<resource>` should rely on version in url. What do you think @xliiv ?

**Comment by szok** [comment]
:ok:

**Comment by mkurek** [comment]
:ok:

**Pull request merged**